### ISSUE TITLE
Amazon S3 library is too old, please update

### ISSUE THREAD

**Issue by hardyoyo** [DSpace/DSpace#8086]
**Describe the bug**
The dependency we use to support an Amazon S3-based assetstore, aws-java-sdk-s3 (1.10.50) is too old, and throws an error (java.lang.IllegalStateException: Socket not created by this factory) if you configure an S3 assetstore. The latest version of the dependency (1.12.129) does not throw this error. The error is mentioned here: https://github.com/aws/aws-sdk-java/issues/1032 and the reported fix was to use a newer version of the library.

We've tested version 1.12.129 of aws-java-sdk-s3 on our DSpace pilot server, Tomcat starts up just fine, and we can deposit items into the repository. We can also confirm those items are going into our Amazon S3 bucket, via the command `dspace bitstore-migrate -p`

```
store[0] == DSBitStoreService, which has 1259 bitstreams.
store[1] == S3BitStoreService, which has 1 bitstreams.
Incoming assetstore is store[1]
```

**To Reproduce**
1. Configure an S3 Assetstore
2. attempt to start your servlet container

**Expected behavior**
When configured with an S3 assetstore, the Servlet container hosting DSpace should start up successfully, and should be usable for uploading new items and bitstreams.

**Related work**
Possibly related to https://github.com/DSpace/DSpace/issues/8078

**Comment by hardyoyo** [comment]
Well... it looks like this commit: https://github.com/DSpace/DSpace/commit/7fca8582604f6ea73d72eb22f1b0f1198b583e15 fixed the issue on master. I should learn to check the master branch before I log issues. :-\ This can be closed, unless we want to bump to the absolute latest of the library?

**Comment by tdonohue** [comment]
Closing, fixed by #8040   If you want to create another small PR for latest version, I'm OK with it. But, I don't see a reason to keep this ticket open waiting for another small version bump if the main bug is fixed.

**Comment by hardyoyo** [comment]
I'll test the version of the dependency that's in master, if it works fine, I'll leave it set to that version.
